azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkInterface.java

```diff
@@ -81,35 +81,42 @@
      * @throws CloudException exceptions thrown from the cloud.
      * @throws IOException exceptions thrown from serialization/deserialization.
      */
-    PublicIpAddress publicIpAddress() throws CloudException, IOException;
+    PublicIpAddress primaryPublicIpAddress() throws CloudException, IOException;
 
     /**
      * @return the resource id of the virtual network subnet associated with this network interface.
      */
-    String subnetId();
+    String primarySubnetId();
```

Comment:
same thing tyo think about - though I know for now returning a Subnet is trickier than returning a PIP, because Subnet is a child, and our model does not yet provide a way to return a child without accessing it via a parent.
So for now, I think this is ok, but I may change this whole approach later - maybe we'll have a better way to deal with references to child objects and objects that can hydrate themselves on demand.